rdr_service/tools/study_nph_biobank_file_export.py

```diff
@@ -0,0 +1,222 @@
+# Script to create biobank file export and store the file in GCS Bucket
+from datetime import datetime, timedelta
+from collections import defaultdict
+from typing import List, Dict, Any, Iterable, Optional
+from zlib import crc32
+from json import dump
+from re import findall
+
+from rdr_service.model.participant_summary import ParticipantSummary as RdrParticipantSummary
+from rdr_service.model.rex import ParticipantMapping as RexParticipantMapping
+from rdr_service.model.study_nph import (
+    Participant as NphParticipant,
+    StudyCategory,
+    Order,
+    OrderedSample,
+    SampleUpdate,
+    BiobankFileExport,
+    SampleExport
+)
+
+from rdr_service.dao.participant_summary_dao import ParticipantSummaryDao as RdrParticipantSummaryDao
+from rdr_service.dao.rex_dao import RexParticipantMappingDao
+from rdr_service.dao.study_nph_dao import (
+    NphParticipantDao,
+    NphStudyCategoryDao,
+    NphOrderDao,
+    NphOrderedSampleDao,
+    NphSampleUpdateDao,
+    NphBiobankFileExportDao,
+    NphSampleExportDao
+)
+
+
+FILE_BUFFER_SIZE_IN_BYTES = 1024 * 1024 # 1MB File Buffer
+
+
+def _filter_orders_by_modified_field(modified_ts: datetime):
+    nph_orders_dao = NphOrderDao()
+    with nph_orders_dao.session() as session:
+        return session.query(Order).filter(
+            Order.modified >= modified_ts
+        ).all()
+
+
+def get_orders_created_or_modified_in_last_n_hours(hours: int = 24) -> Iterable[Order]:
+    modified_ts = datetime.now() - timedelta(hours=hours)
+    return _filter_orders_by_modified_field(modified_ts=modified_ts)
+
+
+def _get_study_category(study_category_id: int) -> StudyCategory:
+    study_category_dao = NphStudyCategoryDao()
+    with study_category_dao.session() as session:
+        return session.query(StudyCategory).get(study_category_id)
+
+
+def _get_parent_study_category(study_category_id: int) -> StudyCategory:
+    study_category = _get_study_category(study_category_id)
+    return _get_study_category(study_category.parent_id)
+
+
+def _get_ordered_samples(order_id: int) -> List[OrderedSample]:
+    ordered_sample_dao = NphOrderedSampleDao()
+    with ordered_sample_dao.session() as session:
+        return session.query(OrderedSample).filter(
+            OrderedSample.order_id == order_id
```

Comment:
This should also filter to only samples that have aliquot ids, to ensure we're only sending the child samples.